dissect/target/filesystems/zip.py

```diff
@@ -0,0 +1,136 @@
+        offset = fh.tell()
+        try:
+            fh.seek(0)
+            zipfile.ZipFile(fileobj=fh, mode="r")
+            return True
+        except Exception as e:
+            log.warning("Failed to detect zip filesystem", exc_info=e)
+            return False
+        finally:
+            fh.seek(offset)
+
+    def get(self, path):
+        """Returns a ZipFilesystemEntry object corresponding to the given path."""
+        return self._fs.get(path)
+
+
+class ZipFilesystemEntry(VirtualFile):
+    def _resolve(self):
+        # NOTE: Normally we would check here if the member is a symlink or not,
+        # however Python does not have symlink support in the zipfile module.
+        # see https://github.com/python/cpython/issues/82102
+        return self
+
+    def open(self):
+        """Returns file handle (file-like object)."""
+        try:
+            f = self.fs.zip.read(self.entry)
+            return BufferedStream(io.BytesIO(f), size=self.entry.file_size)
+        except Exception as e:
+            print(str(e))
+            raise FileNotFoundError()
+
+    def is_dir(self):
+        """Return whether this entry is a directory. Resolves symlinks when possible."""
+        try:
+            return self._resolve().entry.isdir()
+        except FilesystemError:
+            return False
+
+    def is_file(self):
+        """Return whether this entry is a file. Resolves symlinks when possible."""
+        try:
+            return self._resolve().entry.isfile()
+        except FilesystemError:
+            return False
+
+    def is_symlink(self):
+        """Return whether this entry is a link."""
+        # NOTE: Normally we would check here if the member is a symlink or not,
+        # however Python does not have symlink support in the zipfile module.
+        # see https://github.com/python/cpython/issues/82102
+        return False
+
+    def readlink(self):
+        """Read the link if this entry is a symlink. Returns a string."""
+        return self.entry.linkname
+
+    def readlink_ext(self):
+        """Read the link if this entry is a symlink. Returns a filesystem entry."""
+        # Can't use the one in VirtualFile as it overrides the FilesystemEntry
+        return fsutil.resolve_link(fs=self.fs, entry=self)
+
+    def stat(self):
+        """Return the stat information of this entry."""
+        return self._resolve().lstat()
+
+    def lstat(self):
+        """Return the stat information of the given path, without resolving links."""
+        # ['mode', 'addr', 'dev', 'nlink', 'uid', 'gid', 'size', 'atime', 'mtime', 'ctime']
+        return fsutil.stat_result(
+            [
+                self.entry.mode,
+                0,
+                0,
+                0,
+                self.entry.uid,
+                self.entry.gid,
+                self.entry.size,
+                0,
+                self.entry.mtime,
+                0,
+            ]
+        )
```

Comment:
This is not implemented yet, perhaps something like:
```suggestion
        # ['mode', 'addr', 'dev', 'nlink', 'uid', 'gid', 'size', 'atime', 'mtime', 'ctime']
        return fsutil.stat_result(
            [
                (stat.S_IFDIR if self.is_dir() else S_IFREG) | 0o777,
                self.entry.header_offset,
                id(self.fs),
                0,
                0,
                0,
                self.entry.file_size,
                0,
                datetime(*self.entry.date_time, tzinfo=timezone.utc),
                0,
            ]
        )
```